components/Tabbed/index.vue

```diff
@@ -12,16 +12,50 @@ export default {
   },
 
   data() {
-    return { tabs: null };
+    return {
+      tabs:            [],
+      initialTabOrder: []
+    };
+  },
+
+  computed: {
+    filteredTabs() {
+      // keep the tabs list ordered for dynamic tabs
+      const {
+        initialTabOrder,
+        tabs
+      } = this;
+      const out = [];
+
+      if (isEmpty(initialTabOrder) && !isEmpty(tabs)) {
+        tabs.forEach(tab => initialTabOrder.push(tab.name));
+      }
+
+      initialTabOrder.forEach((tabId) => {
+        const match = tabs.find(t => t.name === tabId);
+
+        if (match) {
+          out.push(match);
+        }
+      });
```

Comment:
Yes I see the same thing.

Is there actually a need for truly dynamic (i.e. you don't know how many there are or can be) tabs somewhere?  The usage I know about is like "don't show the X tab if there are no Xes"..  That doesn't need to be implemented as `<Tab v-if="x.length">`, it could just as well always be a child that `<Tabbed>` knows about but have a `visible` prop to control if it's rendered.

If there is actual dynamic usage, then handling just the initial order doesn't sound sufficient.  So rather than trying to keep a list of the "initial" order I would add a weight property to tabs and sort the displayed tabs by that (+label for tiebreaker).

The consuming component can then set the weight explicitly for each tab if it wants.

Or upon (created, and) change of `tabs`, you assign a weight (highest current weight + 1) and set it on any tabs that weren't defined with one.

Reply:
The "WindowManager" for container logs/shell has things that look like tabs and are dynamic, but doesn't use Tabbed.  I think this issue had something to do with that.

There are appearance differences and some extra functionality needed (icon, close button) but I think it could be converted to use Tabbed if you support the fully dynamic case.

Reply:
I did not consider that case of hidden by default. I like the weight idea and will implement that. I had considered the visible prop but this seemed lowest impact. I think its probably the right move though so I'll fix that as well.

Reply:
I refactored out the initial order, added a weight to the `Tab` component which defaults to 0. Now sorting tabs on change with `sortBy` and `'weight' > 'label' > 'name'`. Its way simpler. 

I opted out of refactoring the visible change on the Tab component. I considered using `v-show` but I think that showing/hiding should actually be controlled by `v-if`. `v-if` causes the element, its directives, and components to be destroyed/created on each toggle and this would be the action we want to occur. Any fields that are setup on init or torn down on destroy will run which in the case of adding or removing fields seems desirable.